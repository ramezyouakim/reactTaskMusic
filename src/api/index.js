
import config from '../config'

export default {
    async getFeaturedPlaylists(offset) {
        return fetch(`${config.serverUrl}${config.apiVersion}/browse/featured-playlists?limit=30&offset=${offset}`, {
            method: 'GET',
            headers: new Headers({
                'Authorization': `Bearer ${config.accessToken}`,
                'Content-Type': 'application/json'
            })
        }).then((response) => {
            console.log(response)
            return response
        }).catch((err) => {
            console.log(err)
            return err
        })
    },
    async getArtistPlaylists(playlist_id) {
        return fetch(`${config.serverUrl}${config.apiVersion}/playlists/${playlist_id}/tracks`, {
            method: 'GET',
            headers: new Headers({
                'Authorization': `Bearer ${config.accessToken}`,
                'Content-Type': 'application/json'
            })
        }).then((response) => {
            console.log(response)
            return response
        }).catch((err) => {
            console.log(err)
            return err
        })
    }
}