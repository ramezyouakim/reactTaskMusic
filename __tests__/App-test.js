
import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../src/components/card';
import api from '../src/api';

test('returns result if array', () => {
   api.getFeaturedPlaylists(0)
    .then(((onResponse) => {
      expect(onResponse).toBe((onResponse) => {
        if (typeof onResponse.json().playlists.items == 'array') {
          return true
        }
        else {
          return false
        }
      });
    }))
});

test('card renders correctly', () => {
  const tree = renderer.create(
    <Card
      image={'https://pbs.twimg.com/profile_images/1216813945408966663/vkVajfRz_400x400.jpg'}
      name={'test'}
      owner={'by test'}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});