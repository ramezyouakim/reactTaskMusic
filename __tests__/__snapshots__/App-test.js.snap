// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#fff",
        "margin": 14,
        "marginBottom": 0,
      },
      Object {
        "elevation": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://pbs.twimg.com/profile_images/1216813945408966663/vkVajfRz_400x400.jpg",
      }
    }
    style={
      Array [
        Object {
          "flex": 2,
          "height": 110,
          "resizeMode": "cover",
        },
        Object {
          "display": "flex",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {
          "flex": 2,
          "justifyContent": "center",
          "margin": 20,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "800",
          "marginBottom": 14,
        }
      }
    >
      test
    </Text>
    <Text
      style={Object {}}
    >
      by test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://image.flaticon.com/icons/png/512/16/16049.png",
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "width": 30,
          },
          Object {
            "display": "flex",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#fff",
        "margin": 14,
        "marginBottom": 0,
      },
      Object {
        "elevation": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://pbs.twimg.com/profile_images/1216813945408966663/vkVajfRz_400x400.jpg",
      }
    }
    style={
      Array [
        Object {
          "flex": 2,
          "height": 110,
          "resizeMode": "cover",
        },
        Object {
          "display": "flex",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {
          "flex": 2,
          "justifyContent": "center",
          "margin": 20,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "800",
          "marginBottom": 14,
        }
      }
    >
      test
    </Text>
    <Text
      style={Object {}}
    >
      by test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://image.flaticon.com/icons/png/512/16/16049.png",
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "width": 30,
          },
          Object {
            "display": "flex",
          },
        ]
      }
    />
  </View>
</View>
`;
